superstring/superstring.py

```diff
@@ -70,34 +69,33 @@ def __add__(self, right):
         return SuperStringConcatenation(self, right)
 
     def __str__(self):
-        return self.toPrintable()
+        return self.to_printable()
 
     def __getitem__(self, key):
         # TODO: Negative Indexing
         if isinstance(key, slice):
-            return self.substring(key.start, endIndex = key.stop)
-        return self.characterAt(key)
+            return self.substring(key.start, end_index=key.stop)
+        return self.character_at(key)
+
 
 class SuperString(SuperStringBase):
     def __init__(self, content):
         self._content = content
+        self._length = len(self._content)
```

Comment:
If we look at the `length()` method, it is pretty clear that the idea is to use lazy evaluation, and not bother storing the length unless we actually need it. So this assignment is not needed.

Of course, I suspect that it would be faster to always use `len(self._content)` rather than storing this in a variable, but that's a different topic. :smile:

Reply:
Sorry! That was residual from some other stuff I was doing - let me remove it.